keps/sig-apps/3088-pdb-for-daemonset-pods/README.md

```diff
@@ -0,0 +1,265 @@
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+Support podDusruptionBudget (PDB) for DaemonSet pods. DaemonSet pods are evicted all at once
```

Comment:
Historically PDBs are created to accommodate involuntary disruptions. Reducing the pressure on apiserver has been side effect of the PDBs. Having said that any controller which uses scale sub-resource should be able to leverage the PDB theoritically and since DS controller doesn't have notion of scaling, it perhaps doesn't make sense.

Reply:
Daemonsets are workload controllers and should be able to exert backpressure on eviction.  PDB should not assume scaling, they are about invariants.  So I don't consider scaling to have anything to do with PDB (it's an impl detail)

Reply:
To go back to the original design https://github.com/kubernetes/kubernetes/pull/22217/commits/b77e39298eaf3427bd004402e275dadc3b2e9fbb#diff-716c4ec404aed1a3f24e2c8a19afef4269944ccd97a6721764945f51e0e0e8c5R215 - we talk about pods in aggregate, not controllers.  Very very few components of kube should ever assume anything about a pod based on *which* controller drives it.  They instead deal with pods in aggregate.  We strive to keep controllers decoupled from behavior in order to allow extension of new workload types.